reana_client/cli/files.py

```diff
@@ -341,55 +348,72 @@ def upload_files(ctx, workflow, filenames, access_token):  # noqa: D301
             )
             sys.exit(1)
 
+        try:
+            with open(".gitignore") as f:
+                gitignore_spec = pathspec.PathSpec.from_lines("gitwildmatch", f)
+            display_message(
+                ".gitignore file is detected and will be respected.", msg_type="success"
+            )
+        except FileNotFoundError:
+            gitignore_spec = SimpleNamespace()
+            gitignore_spec.match_file = lambda _: False
+
         if "inputs" in reana_spec:
             filenames = []
+
             filenames += [
                 os.path.join(os.getcwd(), f)
                 for f in reana_spec["inputs"].get("files") or []
             ]
-            filenames += [
+
+            directories_full_path = [
                 os.path.join(os.getcwd(), d)
                 for d in reana_spec["inputs"].get("directories") or []
             ]
-
-    if workflow:
-        if filenames:
-            upload_failed = False
-            for filename in filenames:
-                try:
-                    response = upload_to_server(workflow, filename, access_token)
-                    for file_ in response:
-                        if file_.startswith("symlink:"):
-                            display_message(
-                                "Symlink resolved to {}. "
-                                "Uploaded hard copy.".format(file_[len("symlink:") :]),
-                                msg_type="success",
-                            )
-                        else:
-                            display_message(
-                                "File {} was successfully uploaded.".format(file_),
-                                msg_type="success",
-                            )
-                except FileNotFoundError as e:
-                    logging.debug(traceback.format_exc())
-                    logging.debug(str(e))
+            for directory_path in directories_full_path:
+                for root, dirs, files in os.walk(directory_path):
+                    if not gitignore_spec.match_file(root):
+                        for filename in files:
+                            filename_full_path = os.path.join(root, filename)
+                            if not gitignore_spec.match_file(filename_full_path):
+                                filenames.append(filename_full_path)
```

Comment:
Yes. It would be nice. Do you want to have a new function inside or outside? Also, what would be a good name for it?

For naming, this piece is filtering files inside directories using `.gitignore`, maybe something like "filter_files_with_gitignore"?

Reply:
I think it's alright if the function is inside for now. If we will need to reuse it somewhere else, we will move it.

`filter_files_with_gitignore` is perfect!

Reply:
*SIde note:* function inside does not help with the C901 error.